util_scripts/Builder/Program.cs

```diff
@@ -0,0 +1,145 @@
+        /// <returns>The DTE object that can be used to interact with Visual Studio.</returns>
+        private EnvDTE80.DTE2 getDTE(VSVersion version)
+        {
+            Console.WriteLine("Starting " + version.DTEDesc);
+            Type VSType = System.Type.GetTypeFromProgID(version.DTEDesc);
+            EnvDTE80.DTE2 dte = (EnvDTE80.DTE2)System.Activator.CreateInstance(VSType);
+
+            dte.SuppressUI = true;
+            dte.MainWindow.Visible = false;
+            return dte;
+        }
+
+        /// <summary>
+        /// Builds the specified solution.
+        /// </summary>
+        /// <param name="solution">The solution to build.</param>
+        /// <param name="dte">The DTE (used to gather build errors).</param>
+        /// <returns>true if build was successful, false otherwise</returns>
+        private Boolean buildSolution(Solution solution, EnvDTE80.DTE2 dte) {
+            Console.WriteLine("Started Build");
+            solution.SolutionBuild.Build();
+
+            vsBuildState state = solution.SolutionBuild.BuildState;
+            while (solution.SolutionBuild.BuildState == vsBuildState.vsBuildStateInProgress)
+            {
+                System.Threading.Thread.Sleep(500);
+                state = solution.SolutionBuild.BuildState;
+            }
+
+            Boolean buildSuccess = (solution.SolutionBuild.LastBuildInfo == 0 && state == vsBuildState.vsBuildStateDone);
+
+            dte.ToolWindows.ErrorList.ShowMessages = true;
+            dte.ToolWindows.ErrorList.ShowErrors = true;
+            dte.ToolWindows.ErrorList.ShowWarnings = true;
+
+            Dictionary<vsBuildErrorLevel, String> errorLevel = new Dictionary<vsBuildErrorLevel, String>() { 
+                    {vsBuildErrorLevel.vsBuildErrorLevelHigh, "Error"},
+                    {vsBuildErrorLevel.vsBuildErrorLevelMedium, "Warning"},
+                    {vsBuildErrorLevel.vsBuildErrorLevelLow, "Info"}};
+            ErrorItems errors = dte.ToolWindows.ErrorList.ErrorItems;
+
+            for (int i = 1; i <= errors.Count; i++)
+            {
+                ErrorItem error = errors.Item(i);
+                Console.WriteLine("Build " + errorLevel[error.ErrorLevel] + ": " + error.Description);
+            }
+
+            return buildSuccess;
+        }
+
+        /// <summary>
+        /// Opens a solution and builds it.
+        /// </summary>
+        private void run() {
+            EnvDTE80.DTE2 dte = getDTE(VSVersion.VS_2017);
+
+            Console.WriteLine("Registering message filter");
+            int err = MessageFilter.Register();
+            if (err != 0)
+            {
+                Console.WriteLine("Filter register failed: " + err);
+                this.errorCode = ErrorCode.REGISTER_FAILED;
+            }
+
+            Solution solution = dte.Solution;
+            Console.WriteLine("Opening " + slnPath);
+            solution.Open(slnPath);
+
+            if (buildSolution(solution, dte))

```

Comment:
I got the `build succeeded` message while my TwinCAT installation wasn't properly integrated into my visual studio, is this going to be `True` only when the build has actually worked?

Reply:
I think the issue here is that if something has been opened correctly it will just silently not build it. I've added a check to make sure that at least one twincat project has been loaded.